app/src/main/java/com/myskng/virtualchemlight/activity/SettingsActivity.kt

```diff
@@ -2,35 +2,45 @@ package com.myskng.virtualchemlight.activity
 
 import android.content.SharedPreferences
 import android.os.Bundle
-import android.preference.*
+import android.view.MenuItem
+import androidx.appcompat.app.AppCompatActivity
+import androidx.preference.EditTextPreference
+import androidx.preference.PreferenceFragmentCompat
 import com.myskng.virtualchemlight.R
 
-class SettingsActivity : AppCompatPreferenceActivity() {
+class SettingsActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setupActionBar()
-        fragmentManager.beginTransaction().replace(android.R.id.content, SettingFragment()).commit()
-    }
-
-    private fun setupActionBar() {
         supportActionBar?.setDisplayHomeAsUpEnabled(true)
+        supportFragmentManager.beginTransaction().replace(android.R.id.content, SettingFragment()).commit()
     }
 
-    class SettingFragment : PreferenceFragment(), SharedPreferences.OnSharedPreferenceChangeListener {
-        val sharedPreferences: SharedPreferences by lazy {
-            preferenceManager.sharedPreferences
+    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
+        when (item?.itemId) {
+            android.R.id.home -> {
+                finish()
+                return false
+            }
         }
+        return super.onOptionsItemSelected(item)
+    }
 
-        override fun onCreate(savedInstanceState: Bundle?) {
-            super.onCreate(savedInstanceState)
+    class SettingFragment : PreferenceFragmentCompat(), SharedPreferences.OnSharedPreferenceChangeListener {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/tumugin/VirtualChemLight/blob/62c53567e1253c2a0c6a77e616ca506858ada478/app/src/main/java/com/myskng/virtualchemlight/activity/SettingsActivity.kt#L28"></span>Exceeded max line length (100) (cannot be auto-corrected)</td>
    </tr>
  </tbody>
</table>